cmd/openshift-tuned.go

```diff
@@ -164,51 +202,58 @@ func nodeLabelsRead() map[string]string {
 		if equal := strings.Index(line, "="); equal >= 0 {
 			if key := strings.TrimSpace(line[:equal]); len(key) > 0 {
 				value := line[equal+1:]
-				nodeLabels[key] = value
+				labels[key] = insertUnique(labels[key], value)
 			}
 		} else {
-			/* no '=' sign found */
-			log.Fatalf("Invalid key=value pair in node labels file %s: %s\n", fileNodeLabels, line)
+			// no '=' sign found
+			log.Fatalf("Invalid key=value pair in labels file %s: %s\n", fileLabels, line)
 		}
 	}
+	//
 
-	return nodeLabels
+	return labels
 }
 
-func nodeLabelsDump(nodeLabels map[string]string) {
-	f, err := os.Create(fileNodeLabels)
+func labelsDump(labels map[string][]string, fileLabels string) {
+	f, err := os.Create(fileLabels)
 	if err != nil {
 	}
 	defer f.Close()
 
-	for key, value := range nodeLabels {
-		_, err := f.WriteString(fmt.Sprintf("%s=%s\n", key, value))
-		if err != nil {
-			log.Fatalf("Error writing to node labels file %s: %v\n", fileNodeLabels, err)
+	log.Printf("%s: dumping labels to %s\n", PNAME, fileLabels)
+	for key, values := range labels {
+		for _, value := range values {
+			_, err := f.WriteString(fmt.Sprintf("%s=%s\n", key, value))
+			if err != nil {
+				log.Fatalf("Error writing to labels file %s: %v\n", fileLabels, err)
+			}
 		}
 	}
 	f.Sync()
 }
 
-func nodeLabelsCompare(nodeLabelsOld map[string]string, nodeLabelsNew map[string]string) bool {
-	if nodeLabelsOld == nil {
-		/* no node labels defined yet */
+func labelsCompare(labelsOld map[string][]string, labelsNew map[string][]string, labelType string, fileLabels string) bool {
```

Comment:
Consider having a struct for labels that has a labelType field (and one field with the label map itself).